PingProtect/PingProtect/Sources/Callbacks/PingOneProtectEvaluationCallback.swift

```diff
@@ -0,0 +1,125 @@
+//
+
+import Foundation
+import FRAuth
+
+
+/**
+ * Callback to evaluate Ping One Protect
+ */
+open class PingOneProtectEvaluationCallback: MultipleValuesCallback {
+    
+    /// The pauseBehavioralData received from server
+    public private(set) var pauseBehavioralData: Bool?
+    
+    /// Signals input key in callback response
+    private var signalsKey: String
+    /// Client Error input key in callback response
+    private var clientErrorKey: String
+    
+    /// Designated initialization method for PingOneProtectEvaluationCallback
+    ///
+    /// - Parameter json: JSON object of PingOneProtectEvaluationCallback
+    /// - Throws: AuthError.invalidCallbackResponse for invalid callback response
+    public required init(json: [String : Any]) throws {
+        
+        guard let callbackType = json[CBConstants.type] as? String else {
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        guard let outputs = json[CBConstants.output] as? [[String: Any]], let inputs = json[CBConstants.input] as? [[String: Any]] else {
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        // parse outputs
+        var outputDictionary = [String: Any]()
+        for output in outputs {
+            guard let outputName = output[CBConstants.name] as? String, let outputValue = output[CBConstants.value] else {
+                throw AuthError.invalidCallbackResponse("Failed to parse output")
+            }
+            outputDictionary[outputName] = outputValue
+        }
+        
+        guard let pauseBehavioralData = outputDictionary[CBConstants.pauseBehavioralData] as? Bool else {
+            throw AuthError.invalidCallbackResponse("Missing pauseBehavioralData")
+        }
+        self.pauseBehavioralData = pauseBehavioralData
+        
+        
+        //parse inputs
+        var inputNames = [String]()
+        for input in inputs {
+            guard let inputName = input[CBConstants.name] as? String else {
+                throw AuthError.invalidCallbackResponse("Failed to parse input")
+            }
+            inputNames.append(inputName)
+        }
+        
+        guard let signalsKey = inputNames.filter({ $0.contains(CBConstants.signals) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing signalsKey")
+        }
+        self.signalsKey = signalsKey
+        
+        guard let clientErrorKey = inputNames.filter({ $0.contains(CBConstants.clientError) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing clientErrorKey")
+        }
+        self.clientErrorKey = clientErrorKey
+        
+        try super.init(json: json)
+        type = callbackType
+        response = json
+    }
+    
+    
+    /// Get Signals from the device
+    /// - Parameter completion: Completion block for signals result
+    open func getSignals(completion: @escaping ProtectResultCallback) {
```

Comment:
I noticed you have `getSignals` intermixed with `getData`. Is this intentional?

Reply:
As the input name in our Callback is called **signals**, I called this method `getSignals` instead of `getData`. Not apposed to calling it `getData` either.